From ca58934537ef3695f78c40d6fe5d7acbc837a674 Mon Sep 17 00:00:00 2001
From: Myron Stowe <myron.stowe@redhat.com>
Date: Thu, 10 Jul 2014 01:08:33 -0400
Subject: [acpi] hotplug: Do not execute "insert in progress" _OST

Message-id: <20140710010833.22146.47848.stgit@gir.stowe>
Patchwork-id: 85637
O-Subject: [RHEL7.1 PATCH 098/105] ACPI / hotplug: Do not execute "insert in progress" _OST
Bugzilla: 1114228
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1114228
Upstream Status: 176a88d79d6b5aebabaff16734e8b3107efcaaad

commit 176a88d79d6b5aebabaff16734e8b3107efcaaad
Author: Rafael J. Wysocki <rafael.j.wysocki@intel.com>
Date:   Thu Nov 7 01:42:17 2013 +0100

    ACPI / hotplug: Do not execute "insert in progress" _OST

    According to the ACPI spec (5.0, Section 6.3.5), the "Device
    insertion in progress (pending)" (0x80) _OST status code is
    reserved for the "Insertion Processing" (0x200) source event
    which is "a result of an OSPM action".  Specifically, it is not
    a notification, so that status code should not be used during
    notification processing, which unfortunately is done by
    acpi_scan_bus_device_check().

    For this reason, drop the ACPI_OST_SC_INSERT_IN_PROGRESS _OST
    status evaluation from there (it was a mistake to put it in there
    in the first place).

    Signed-off-by: Rafael J. Wysocki <rafael.j.wysocki@intel.com>
    Cc: All applicable <stable@vger.kernel.org>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/acpi/scan.c b/drivers/acpi/scan.c
index 7093a58..96a5052 100644
--- a/drivers/acpi/scan.c
+++ b/drivers/acpi/scan.c
@@ -343,8 +343,6 @@ static void acpi_scan_bus_device_check(acpi_handle handle, u32 ost_source)
    goto out;
   }
  }
- acpi_evaluate_hotplug_ost(handle, ost_source,
-      ACPI_OST_SC_INSERT_IN_PROGRESS, NULL);
  error = acpi_bus_scan(handle);
  if (error) {
   acpi_handle_warn(handle, "Namespace scan failure\n");
-- 
1.7.1